nuklear_console_button.h

```diff
@@ -195,17 +196,38 @@ NK_API struct nk_rect nk_console_button_render(nk_console* console) {
     return widget_bounds;
 }
 
+static void nk_console_button_destroy(nk_console* button, void* unused) {
+    NK_UNUSED(unused);
+    nk_console_button_data* data = (nk_console_button_data*)button->data;
+    if (data == NULL) {
+        return;
+    }
+    nk_console_event_handler_destroy(button, &data->onclick);
+}
+
 /**
  * Create a button.
  */
 NK_API nk_console* nk_console_button(nk_console* parent, const char* text) {
-    return nk_console_button_onclick(parent, text, NULL);
+    // Create the widget data.
+    nk_console_button_data* data = (nk_console_button_data*)NK_CONSOLE_MALLOC(nk_handle_id(0), NULL, sizeof(nk_console_button_data));
+    nk_zero(data, sizeof(nk_console_button_data));
+
+    nk_console* button = nk_console_label(parent, text);
+    button->type = NK_CONSOLE_BUTTON;
+    button->data = (void*)data;
+    button->selectable = nk_true;
+    button->columns = 1;
+    button->render = nk_console_button_render;
+    button->destroy = &nk_console_button_destroy;
```

Comment:
Went through a review of this, and it's looking solid. Question...

The Button's `destroy` event is used in `nuklear_console_file.h` to clean up the file entries for each individual file. Is this a problem?
https://github.com/RobLoach/nuklear_console/blob/601a3e2cb646257fbe0e9e2a8e0a4aedf0e40c0e/nuklear_console_file.h#L285

I guess `onchange` isn't used in these, so it doesn't really matter?

Reply:
Good catch, I didn't notice that. Yeah ideally the button destructor wouldn't be blindly overridden and instead the old destructor should be called after the new one. This is an example where your suggestion to have a list of callbacks could be helpful.

In this particular example I think you're right that it's fine because the buttons are created implicitly and therefore don't have on change handlers. Should be fine to leave it for now but I'd suggest a follow up PR to make this more robust. If you're good with that I can try to have a follow up PR ready in the next couple days (or happy to review one from you!). Or lmk if you'd prefer I improve it before merging this PR

Reply:
Yeah, I think if we have a `nk_console_event_handler* events` cvector, it could handle all the event types, along with future event types we want to introduce. Excluding `render()` for performance reasons, all of the following could be handled by the list...

- onchange
- destroy
- onclick

Going to merge this one for now. Thanks so much!